File: real estate web/FE/src/app/service/data-form.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {DataForm} from '../entity/form/data-form';

@Injectable({
  providedIn: 'root'
})
export class DataFormService {
  private url = 'http://localhost:8080/api/form';
  httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json'
    })
  };

  constructor(private httpClient: HttpClient) {
  }
  /**
   * Create by: KhanhLB
   * Date created: 03/02/2023
   * Function: get list dataForm from BE
   * @param contentDataForm,page
   * @return pageDataForm
   */
  searchByContent(contentDataForm: string, page: number): Observable<any> {
    if (contentDataForm === '') {
      return this.httpClient.get<any>('http://localhost:8080/api/form?page=' + page);
    } else {
      return this.httpClient.get<any>('http://localhost:8080/api/form?contentDataForm=' + contentDataForm + '&page=' + page);
    }
  }

  /**
   * Create bt: KhanhLB
   * Date created: 03/02/2023
   * Function: save dataForm in database
   * @param: dataForm
   */
  createDataFormDTO(dataForm: DataForm): Observable<DataForm> {
    return this.httpClient.post<DataForm>('http://localhost:8080/api/form/save', JSON.stringify(dataForm), this.httpOptions);
  }

  findById(id: number): Observable<DataForm> {
    return this.httpClient.get<DataForm>(this.url + '/' + id);
  }


  updateDataForm(dataForm: DataForm): Observable<DataForm> {

    return this.httpClient.put<DataForm>(this.url + '/update/' + dataForm.idDataForm, dataForm);
  }


  deleteById(id: any): Observable<DataForm> {
    return this.httpClient.delete<DataForm>(this.url + '/delete/' + id);
  }
}

File: real estate web/BE/src/main/java/com/c0722g1repobe/service/form/impl/DataFormService.java
package com.c0722g1repobe.service.form.impl;

import com.c0722g1repobe.entity.form.DataForm;
import com.c0722g1repobe.repository.form.IDataFormRepository;
import com.c0722g1repobe.service.form.IDataFormService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class DataFormService implements IDataFormService {
    @Autowired
    private IDataFormRepository iDataFormRepository;
    /**
     * Create by: KhanhLB
     * Date created: 31/01/2023
     * Function: show list dataForm
     *
     * @param contentDataForm,pageable
     * @return json list dataForm
     */
    @Override
    public Page<DataForm> searchByContent(String contentDataForm, Pageable pageable) {
        return iDataFormRepository.searchByContent(contentDataForm,pageable);
    }
    /**
     * Create by: KhanhLB
     * Date created: 31/01/2023
     * Function: save dataForm
     *
     * @param contentDataForm,urlDataForm
     * @return json list dataForm
     */
    @Override
    public void saveDataForm(String contentDataForm, String urlDataForm) {
        iDataFormRepository.saveDataForm(contentDataForm,urlDataForm);
    }
    @Override
    public void deleteByIdDataForm(long id) {
        iDataFormRepository.deleteByIdDataForm(id);
    }


    @Override
    public DataForm findByIdDataForm(long id) {
        return iDataFormRepository.findByIdDataForm(id);
    }

    @Override
    public void updateDataForm(DataForm dataForm) {
        iDataFormRepository.updateDataForm( dataForm.getContentDataForm(), dataForm.getUrlDataForm(), dataForm.getIdDataForm());
    }
}
